[partitioner] Fix argument to _broadcast_on_rank0

fmassa opened the pull request:
Summary:
There was a bug when I refactored my original implementation.

This should fix it

Test Plan: Run on some internal workloads

Differential Revision: D74190485

danthe3rd reviewed (approved):
Approving as it's already approved in the internal review by @bdhirsh

fmassa committed:
[partitioner] Fix argument to _broadcast_on_rank0 (#152846)

Summary:

There was a bug when I refactored my original implementation.

This should fix it

Test Plan: Run on some internal workloads

Reviewed By: bdhirsh, Microve, danthe3rd

Differential Revision: D74190485

fmassa commented:
@pytorchbot merge
